.circleci/config.yml

```
@@ -1,70 +1,168 @@
+                wget https://dl.google.com/go/go1.13.4.linux-amd64.tar.gz
+          - run:
+              name: "Check downloaded Go checksum"
+              command: |
+                [[ "$(sha256sum go1.13.4.linux-amd64.tar.gz | cut -d ' ' -f 1)" == "692d17071736f74be04a72a06dab9cac1cd759377bd85316e52b2227604c004c" ]]
+          - run:
+              name: "Install Go"
+              command: |
+                sudo tar -C /usr/local -xzf go1.13.4.linux-amd64.tar.gz
+          - run:
+              name: "Set Go environment"
+              command: |
+                go env -w GOBIN=/home/circleci/.go_workspace/bin
+          - run:
+              name: "Create $GOPATH directory"
+              command: |
+                sudo mkdir $(go env GOPATH) && sudo chown $USER $(go env GOPATH)
+          - run:
+              name: "Remove downloaded Go files"
+              command: |
+                rm go1.13.4.linux-amd64.tar.gz
+          - run: 
+              name: "Check Go version"
+              command: |
+                go version
+
+      install-tools:
+        steps:
+          - run:
+              name: "Install kind"
+              command: |
+                CGO_ENABLED=0 GO111MODULE="on" go get sigs.k8s.io/kind@v0.7.0
+          - run:
+              name: "Install kubectl"
+              command: |
+                curl -LO https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl
+                chmod +x ./kubectl
+                sudo mv ./kubectl /usr/local/bin/kubectl
+                kubectl version --client
+
+      create-cluster:
+        parameters:
+          kind-config:
+            type: string
+        steps:
+          - run:
+              name: "Create cluster"
+              command: |
+                cd ~/project
+                
+                if [ -e << parameters.kind-config >> ]; then
+                  cp << parameters.kind-config >> kind-config.yaml
+                else
+                  echo "kind: Cluster" >> kind-config.yaml
+                  echo "apiVersion: kind.sigs.k8s.io/v1alpha3" >> kind-config.yaml
+                fi
+
+                kind create cluster --config kind-config.yaml --image quay.io/giantswarm/kind-node:$KUBERNETES_VERSION
+
+      test-setup:
+        parameters:
+          setup-script:
+            type: string
+        steps:
+          - run:
+              name: "Run test setup commands"
+              command: |
+                if [ -e << parameters.setup-script >> ]; then
+                  echo << parameters.setup-script >>
+                  sh << parameters.setup-script >>
+                fi
+
+      go-test:
+        parameters:
+          env:
+            type: string
+          test-dir:
+            type: string
+        steps:
+          - run:
+              name: "Create .env file"
+              command: |
+                echo '<< parameters.env >>' | sed -e 's/^\(.\)/export \1/' > .env
```

I recently realised using `tee` for those things is very useful for debugging.

```suggestion
                echo '<< parameters.env >>' | sed -e 's/^\(.\)/export \1/' | tee .env
```